api/v1alpha1/nifinodegroupautoscaler_types.go

```diff
@@ -32,16 +32,18 @@ type NifiNodeGroupAutoscalerSpec struct {
 	// +kubebuilder:validation:Pattern:="[a-z0-9]([-a-z0-9]*[a-z0-9])?"
 	// +kubebuilder:validation:MaxLength:=63
 	NodeConfigGroupId string `json:"nodeConfigGroupId"`
-	// A label selector used to identify & manage Node objects in the referenced NifiCluster. Any node matching this selector will be managed by this autoscaler. Even if that node was previously statically defined.
+	// A label selector used to identify & manage Node objects in the referenced NifiCluster. Any node matching this
+	// selector will be managed by this autoscaler. Even if that node was previously statically defined.
 	NodeLabelsSelector *metav1.LabelSelector `json:"nodeLabelsSelector"`
 	// the node readOnlyConfig for each node in the node group
 	// +optional
 	ReadOnlyConfig *ReadOnlyConfig `json:"readOnlyConfig,omitempty"`
-	// the nodeConfig to use for each node in the node group. This will be merged with and is preferred to the configured nodeConfigGroupId
+	// the nodeConfig to use for each node in the node group. This will be merged with and is preferred to the configured
+	// nodeConfigGroupId
 	// +optional
 	NodeConfig *NodeConfig `json:"nodeConfig,omitempty"`
 	// current number of replicas expected for the node config group
-	// +kubebuilder:default:=1
+	// +kubebuilder:default:=0
```

Comment:
This change just ensures that no pods get spun up in the autoscale node group until the HPA decided to scale up, correct?

It's worth mentioning that HPA scaling to zero isn't a GA feature in Kubernetes, yet. It is currently [feature gated](https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates/) as of Kubernetes 1.16. The feature gate is `HPAScaleToZero` and is by default false (not enabled).

For clusters that don't have this feature gate enabled, will this change cause a problem?

Reply:
I assume this will not, because if you configure an HPA without the feature gate enabled, it will not allow you to set the `minReplicas` to `0` so it will at least be set to `1` and automatically change the `replica` field of the `NifiNodeGroupAutoScaler` resource to `1`

Reply:
I think so too, but I thought it was worth asking!